aiohttp/web_exceptions.py

```diff
@@ -427,11 +427,16 @@ def __init__(
         text: Optional[str] = None,
         content_type: Optional[str] = None,
     ) -> None:
+        if not link:
+            raise ValueError(
+                "HTTP unavailable for legal reasons needs a link to"
+                " a resource with information for the blocking reason."
+            )
```

Comment:
I think you've misread the spec:

> When an entity blocks access to a resource and returns status 451, it SHOULD include a "Link" HTTP header field [[RFC5988](https://datatracker.ietf.org/doc/html/rfc5988)] whose value is a URI reference [[RFC3986](https://datatracker.ietf.org/doc/html/rfc3986)] identifying itself.  When used for this purpose, the "Link" header field MUST have a "rel" parameter whose value is "blocked-by".
> 
> The intent is that the header be used to identify the entity actually implementing blockage, not any other entity mandating it.  A human-readable response body, as discussed above, is the appropriate location for discussion of administrative and policy issues.

So, we should probably allow `None` and then exclude the header in that case. The link is also to identify who is implementing the blocking, not the reason for the blocking (which should be in the body).

Reply:
So, I would suggest changing the parameter to allow `None`, and then only add the header under `if link:`. Plus, update the docs below to give the correct usecase for a link.

If you're happy to do that, I think it'll be good to merge.

Reply:
Oh, you are absolutely right.
I did read that a link is optional when reviewing the spec but then my brain just forgot it when implementing 🤦.
Regarding the link description, I just didn't want to change it too much from the existing description.
I will do all the suggested changes 👍.

Reply:
Implemented change in [8fde58b](https://github.com/aio-libs/aiohttp/pull/7689/commits/8fde58b3499cd3310f7b790e6a51b181e9bca75d).
Let me know if anything else is needed.